### ISSUE TITLE
Look into using Node's built-in test runner

### ISSUE THREAD

**Issue by matt-boris** [taskcluster/taskcluster#6670]
Once https://github.com/taskcluster/taskcluster/issues/6647 is closed as complete, I think we should look into migrating to node's (now stable as of v20) test-runner.

It's likely faster than mocha and it's all built-in, so no dependencies, which means we'd be able to remove a bunch of node deps.

Test Runner API docs: https://nodejs.org/dist/latest-v20.x/docs/api/test.html

**Comment by Eijebong** [comment]
I've PoC'ed 2 migrations here: https://github.com/Eijebong/taskcluster/commit/14992b61a36890d50883c27a55c89d3d949314c6

Things I've learned:

- Migration is mostly mechanical (i.e replacing `suite` -> `describe`, `test` -> `it`). That's the good and easy part
- It would remove ~300 lines of lockfile (out of 15k) but most of them (~25) are only used by mocha, which would be nice to have for sure
- You will run into node --test issues regarding global setup/teardown because that's not a concept there and has to be brought per suite.
- You will run into issues for some things that are not process safe. (i.e the hardcoded port in my PoC that I had to remove because node --test runs tests in parallel in multiple processes, isolating them by file by default).
- It's not faster. The `iterate` tests are fairly similar in time but take slightly more CPU. The `api` tests are ~5-10% faster but take **8x** more CPU time. It's possible to disable parallelism and get back in the same ballpark with time/CPU usage but then what's the point...
- `libraries/testing` is going to be the major pain of this.
I would not recommend putting more time into this right now unless mocha becomes a pain to update/maintain.

**Comment by matt-boris** [comment]
Thanks @Eijebong for looking into this and PoC'ing! I'm fine closing this for now then since it doesn't seem worthwhile for us.
